pkg/controller/ads/ads_processor.go

```diff
@@ -140,13 +144,20 @@ func (svc *processor) handleCdsResponse(resp *service_discovery_v3.DiscoveryResp
 	current := sets.New[string]()
 	lastEdsClusterNames := svc.Cache.edsClusterNames
 	svc.Cache.edsClusterNames = []string{}
+	svc.Cache.dnsClusters = make(map[string]time.Duration)
 	for _, resource := range resp.GetResources() {
 		if err = anypb.UnmarshalTo(resource, cluster, proto.UnmarshalOptions{}); err != nil {
 			continue
 		}
 		current.Insert(cluster.GetName())
 		if cluster.GetType() == config_cluster_v3.Cluster_EDS {
 			svc.Cache.edsClusterNames = append(svc.Cache.edsClusterNames, cluster.GetName())
+		} else if cluster.GetType() == config_cluster_v3.Cluster_STRICT_DNS ||
+			cluster.GetType() == config_cluster_v3.Cluster_LOGICAL_DNS {
+			clusterNames := getDoaminAndRefreshRateFromCluster(cluster)
```

Comment:
Please separate the return value, one is a slice of doamins, the other refreshRate